executors/cmdexec/volume.go

```diff
@@ -276,6 +276,35 @@ func (s *CmdExecutor) VolumeInfo(host string, volume string) (*executors.Volume,
 	return &volumeInfo.VolInfo.Volumes.VolumeList[0], nil
 }
 
+func (s *CmdExecutor) VolumeStatusDetailed(host string, volume string) (*executors.VolumeDetail, error) {
+	godbc.Require(volume != "")
+	godbc.Require(host != "")
+
+	type CliOutput struct {
+		OpRet     int                 `xml:"opRet"`
+		OpErrno   int                 `xml:"opErrno"`
+		OpErrStr  string              `xml:"opErrstr"`
+		VolStatus executors.VolStatus `xml:"volStatus"`
+	}
+
+	command := []string{
+		fmt.Sprintf("gluster --mode=script volume status %v detail --xml", volume),
+	}
+
+	//Get the xml output of volume info
+	output, err := s.RemoteExecutor.RemoteCommandExecute(host, command, 10)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to get detailed volume status of volume name: %v", volume)
+	}
+	var volumeDetails CliOutput
+	err = xml.Unmarshal([]byte(output[0]), &volumeDetails)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to determine detailed volume status of volume name: %v", volume)
+	}
+	logger.Debug("%+v\n", volumeDetails)
```

Comment:
Please verify volumeDetails.opRet as well.

Reply:
I can check it if needed. But neighbour method `VolumeInfo` also not performs this check.